prqlc/prqlc/src/cli.rs

```diff
@@ -202,23 +203,24 @@ impl Command {
             Command::Format { input } => {
                 let sources = read_files(input)?;
 
-                if sources.sources.len() != 1 {
-                    let paths = sources
-                        .sources
-                        .keys()
-                        .map(|x| format!("`{}`", x.display()))
-                        .sorted()
-                        .join(", ");
-                    bail!(
-                        "Currently `fmt` only works with a single source, but found multiple sources: {paths:?}"
-                    )
-                }
-                let (_, source) = sources.sources.into_iter().next().unwrap();
+                for (path, source) in sources.sources {
```

Comment:
@aljazerzen do you think we should add the project root to the `SourceTree` struct? Currently it doesn't know the root and so can't write to the correct location.

Or is there a better data structure to use?

Reply:
<!--
So, the idea was for `SourceTree` to a mapping from paths to file contents. So path of each file is already in the mapping.

But then there was a problem that when invoking a single file, that file should be treated as the root of the project, regardless of it's name. So I've made it so that it has the path "". I suppose this is causing problems now.
-->

`SourceTree` is the data structure that is supposed to pass all needed sources to the compiler, along with their hierarchy. It would make sense to also information to obtain the absolute path to each of the files, so this can be displayed in error messages for example, or be used when formatting the files. 

So yes, an absolute path to the directory containing all the source files is a good idea. And it should be `Option<PathBuf>`, as this does not make sense when compiling a single snippet.

Reply:
One alternative would be to store absolute paths in `sources` — WDYT?

Reply:
(implemented with the additional `root` struct member, but can refactor later if needed)

Reply:
I'm uneasy with absolute paths in sources, because then the mapping looks like it has the root at `/`. Because of that the resolver would have to find the longest common prefix in all paths to determine the root.

So current state better.